src/ice_shelf/MOM_ice_shelf_dynamics.F90

```diff
@@ -1940,18 +1953,22 @@ subroutine calc_shelf_driving_stress(CS, ISS, G, US, taudx, taudy, OD)
   rhoi_rhow = rho/rhow
   ! prelim - go through and calculate S
 
-  S(:,:) = -CS%bed_elev(:,:) + ISS%h_shelf(:,:)
-  ! check whether the ice is floating or grounded
+  if (CS%GL_couple) then
+    S(:,:) = -CS%bed_elev(:,:) + OD(:,:) + ISS%h_shelf(:,:)
```

Comment:
Please revise this line to avoid array syntax working on halo regions that might have NaNs, and also please add parentheses to specify the order with which these three arrays should be added.

Reply:
I made these changes, and the similar ones we discussed for PR #511